pkgs/applications/graphics/structorizer/default.nix

```diff
@@ -0,0 +1,105 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, jdk11
+, makeDesktopItem
+, makeWrapper
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "structorizer";
+  version = "3.32-11";
+
+  desktopItems = [
+    (makeDesktopItem {
+      type = "Application";
+      name = "Structorizer";
+      desktopName = "Structorizer";
+      genericName = "Diagram creator";
+      comment = meta.description;
+      icon = pname;
+      exec = pname;
+      terminal = false;
+      mimeTypes = [ "application/nsd" ];
+      categories = [
+        "Development"
+        "Graphics"
+        "VectorGraphics"
+        "RasterGraphics"
+        "ComputerScience"
+      ];
+      keywords = [ "nsd" "diagrams" ];
+    })
+  ];
+
+  src = fetchFromGitHub {
+    owner = "fesch";
+    repo = "Structorizer.Desktop";
+    rev = version;
+    hash = "sha256-rGyeOcGm6uBplgTjMIOy/xRekfHacwDy9kkMigmRSdk=";
+  };
+
+  nativeBuildInputs = [ jdk11 makeWrapper copyDesktopItems ];
```

Comment:
not sure about this one, but since jdk11 is also a runtime dependency, shouldn't it go in `buildInputs`?

Reply:
I'm confused by the different build inputs but it still works with `strictDeps = true`.

Reply:
https://nixos.org/manual/nixpkgs/unstable/#ssec-stdenv-dependencies-overview agrees with you so I guess `strictDeps = true` might not do what I think it does or it doesn't catch it. I'm putting the jdk into `buildInputs`.

Reply:
Also, I'm not using a jre for runtime because there is no jre11 that runs on systems not supporting OpenJDK (I think).